python/dgl/graphbolt/impl/fused_csc_sampling_graph.py

```diff
@@ -212,12 +214,39 @@ def node_type_offset(self) -> Optional[torch.Tensor]:
         """
         return self._c_csc_graph.node_type_offset()
 
+    @property
+    def node_type_offset_list(self) -> Optional[list]:
```

Comment:
I don't think we need a dedicated method to obtain the list version of `node_type_offset`. user is free to call `node_type_offset = g.node_type_offset.tolist()` if `tensor` is not what wanted.

Reply:
I think we'd better define the required APIs only to keep the API set to be concise.

Reply:
Can we make it a private property and call it `_node_type_offset_list`? We basically need a guaranteed CPU version of `node_type_offset` for use in GraphBolt internals.

Reply:
For performance reasons.

Reply:
why do we need a guaranteed CPU version(list?)? we're free to call `tolist()` instead of `.item()` internally.

Reply:
When graph is moved to GPU, node type offset is moved as well. However, when we access it in graphbolt internals, it triggers CPU GPU copies that are totally unnecessary and hurts performance.

Reply:
All modifications in this PR are examples of CPU accesses to node type offset.

Reply:
If we call tolist when we access, we copy it at every minibatch train iteration. Introducing this property let's us cache the copied version.

Reply:
ok, then let's make it private for now.

Reply:
Pushed the private version.